src/Handler/CurlFactory.php

```diff
@@ -172,13 +176,22 @@ private static function createRejection(EasyHandle $easy, array $ctx)
                 )
             );
         }
-
-        $message = sprintf(
-            'cURL error %s: %s (%s)',
-            $ctx['errno'],
-            $ctx['error'],
-            'see https://curl.haxx.se/libcurl/c/libcurl-errors.html'
-        );
+        if (version_compare($ctx[self::CURL_VERSION_STR], self::LOW_CURL_VERSION_NUMBER)) {
```

Comment:
Sorry to be late but this seems wrong, it's missing a comparison operator (either against 0 or as a third string argument, see [examples](https://www.php.net/manual/en/function.version-compare.php#refsect1-function.version-compare-examples)). Currently it's basically testing `curl_version()['version'] != '7.21.2'`. I guess you intended `>=` here. But actually I couldn't find any evidence that curl 7.21.2+ added the URI in the error message... Could you please explain your scenario?

Reply:
ping @sagikazarmark: the code needs to be fixed!
Either you have evidence that curl 7.21.2+ added the URI in the error message, and then you should add `>= 0` (or `> 0`) between `))`;
or you don't, and then this should be reverted, and maybe re-submit only https://github.com/guzzle/guzzle/pull/2108/commits/bd368857d54081a9eeaea3c0ecfcb4e10455e56b

Reply:
proposed #2648